### DISCUSSION TITLE
Multi-threaded emulation on or off?

### DISCUSSION THREAD

**Discussion by kathyrollo** [flyinghead/flycast#1371] [Q&A] [answered]
Hi @flyinghead 

I was going through the Advanced settings tab and found multi-threaded emulation disabled. When would you generally recommend turning this particular setting on?

![Screenshot 2024-01-18 160144](https://github.com/flyinghead/flycast/assets/28589393/a9ff3e85-f3fb-4f99-bf56-323923fd4ecb)

**Comment by vkedwardli** [comment, marked as answer]
The original hardware is closer to multi-threaded way since we have both the CPU+GPU working in parallels, however the emulation is not perfect sometimes: e.g. there is issue like race condition for texture update / off canvas rendering.

Some say single threaded rendering provides a better frame pacing or less input lag.

For me, if I have a super powerful hardware, I would use single-threaded mode, vice versa.

YMMV.

**Reply by kathyrollo** [reply]
Hi, would you say a Ryzen 7 5800H (8C/16T) is powerful enough for single-threading?

**Reply by vkedwardli** [reply]
Yes it's a very modern CPU, as long as you don't push too far on the graphics, it can handle single threaded mode flawlessly

You could also run multi vs single and see if you can feel any difference

**Reply by kathyrollo** [reply]
Thanks, interested to hear your thoughts on this. Given your overall experience, would you consider sticking to single-threading for emulators in general if there's a powerful-enough CPU?

Context being, I noticed other emulators have a similar multi-core/threading option and some games only work when the option is turned off. Was wondering if it's worth turning it off for all emulators, only enabling on a game-specific basis.

**Reply by vkedwardli** [reply]
Race condition is a bitch and it is hard to emulate it flawlessly with multi-threading mode, but usually i go for the "stick to the default" way, if the emulator is default for multi-threaded, I won't change to single threaded on purpose unless I face problems

**Reply by kathyrollo** [reply]
Sounds reasonable, thank you.
